lib/awesome_spawn.rb

```diff
@@ -61,9 +61,8 @@ def logger
   # @return [CommandResult] the output stream, error stream, and exit status
   # @see http://ruby-doc.org/core/Kernel.html#method-i-spawn Kernel.spawn
   def run(command, options = {})
-    raise ArgumentError, "options cannot contain :out" if options.include?(:out)
-    raise ArgumentError, "options cannot contain :err" if options.include?(:err)
-    raise ArgumentError, "options cannot contain :in" if options.include?(:in)
+    bad_key = flat_keys(options).detect { |fd| [:in, :out, :err].include?(fd) }
```

Comment:
I think this might be simpler (to read at least)

```ruby
bad_keys = flat_keys(options) & [:in, :out, :err]
raise ArgumentError, "options cannot contain #{bad_keys}" if bad_keys.any?
```

If you drop the flat_keys method with my other suggestion, then this becomes

```ruby
bad_keys = options.keys.flatten & [:in, :out, :err]
raise ArgumentError, "options cannot contain #{bad_keys}" if bad_keys.any?
```